From 963519ce88e572c3e21e11819b2a3d535d1fa979 Mon Sep 17 00:00:00 2001
From: Mahesh Mahadevan <r9aadq@freescale.com>
Date: Mon, 31 Oct 2011 08:36:05 -0500
Subject: [PATCH 0515/2463] ENGR00161119 Codec regulator registration update

Move the codec regulator registration code to the board
specific file

Signed-off-by: Mahesh Mahadevan <r9aadq@freescale.com>
---
 arch/arm/mach-mx6/board-mx6q_arm2.c      |   49 ++++++++++++++++--------------
 arch/arm/mach-mx6/mx6_anatop_regulator.c |    6 ----
 2 files changed, 26 insertions(+), 29 deletions(-)

diff --git a/arch/arm/mach-mx6/board-mx6q_arm2.c b/arch/arm/mach-mx6/board-mx6q_arm2.c
index 9d3e982..a01c7d3 100644
--- a/arch/arm/mach-mx6/board-mx6q_arm2.c
+++ b/arch/arm/mach-mx6/board-mx6q_arm2.c
@@ -883,84 +883,83 @@ static struct imx_esai_platform_data sab_esai_pdata = {
 
 #ifdef CONFIG_SND_SOC_SGTL5000
 
-static struct regulator_consumer_supply sgtl5000_consumer_vdda = {
+static struct regulator_consumer_supply sgtl5000_arm2_consumer_vdda = {
 	.supply = "VDDA",
 	.dev_name = "0-000a",
 };
 
-static struct regulator_consumer_supply sgtl5000_consumer_vddio = {
+static struct regulator_consumer_supply sgtl5000_arm2_consumer_vddio = {
 	.supply = "VDDIO",
 	.dev_name = "0-000a",
 };
 
-static struct regulator_consumer_supply sgtl5000_consumer_vddd = {
+static struct regulator_consumer_supply sgtl5000_arm2_consumer_vddd = {
 	.supply = "VDDD",
 	.dev_name = "0-000a",
 };
 
-static struct regulator_init_data sgtl5000_vdda_reg_initdata = {
+static struct regulator_init_data sgtl5000_arm2_vdda_reg_initdata = {
 	.num_consumer_supplies = 1,
-	.consumer_supplies = &sgtl5000_consumer_vdda,
+	.consumer_supplies = &sgtl5000_arm2_consumer_vdda,
 };
 
-static struct regulator_init_data sgtl5000_vddio_reg_initdata = {
+static struct regulator_init_data sgtl5000_arm2_vddio_reg_initdata = {
 	.num_consumer_supplies = 1,
-	.consumer_supplies = &sgtl5000_consumer_vddio,
+	.consumer_supplies = &sgtl5000_arm2_consumer_vddio,
 };
 
-static struct regulator_init_data sgtl5000_vddd_reg_initdata = {
+static struct regulator_init_data sgtl5000_arm2_vddd_reg_initdata = {
 	.num_consumer_supplies = 1,
-	.consumer_supplies = &sgtl5000_consumer_vddd,
+	.consumer_supplies = &sgtl5000_arm2_consumer_vddd,
 };
 
-static struct fixed_voltage_config sgtl5000_vdda_reg_config = {
+static struct fixed_voltage_config sgtl5000_arm2_vdda_reg_config = {
 	.supply_name		= "VDDA",
 	.microvolts		= 1800000,
 	.gpio			= -1,
-	.init_data		= &sgtl5000_vdda_reg_initdata,
+	.init_data		= &sgtl5000_arm2_vdda_reg_initdata,
 };
 
-static struct fixed_voltage_config sgtl5000_vddio_reg_config = {
+static struct fixed_voltage_config sgtl5000_arm2_vddio_reg_config = {
 	.supply_name		= "VDDIO",
 	.microvolts		= 3300000,
 	.gpio			= -1,
-	.init_data		= &sgtl5000_vddio_reg_initdata,
+	.init_data		= &sgtl5000_arm2_vddio_reg_initdata,
 };
 
-static struct fixed_voltage_config sgtl5000_vddd_reg_config = {
+static struct fixed_voltage_config sgtl5000_arm2_vddd_reg_config = {
 	.supply_name		= "VDDD",
 	.microvolts		= 0,
 	.gpio			= -1,
-	.init_data		= &sgtl5000_vddd_reg_initdata,
+	.init_data		= &sgtl5000_arm2_vddd_reg_initdata,
 };
 
-struct platform_device sgtl5000_vdda_reg_devices = {
+static struct platform_device sgtl5000_arm2_vdda_reg_devices = {
 	.name	= "reg-fixed-voltage",
 	.id	= 0,
 	.dev	= {
-		.platform_data = &sgtl5000_vdda_reg_config,
+		.platform_data = &sgtl5000_arm2_vdda_reg_config,
 	},
 };
 
-struct platform_device sgtl5000_vddio_reg_devices = {
+static struct platform_device sgtl5000_arm2_vddio_reg_devices = {
 	.name	= "reg-fixed-voltage",
 	.id	= 1,
 	.dev	= {
-		.platform_data = &sgtl5000_vddio_reg_config,
+		.platform_data = &sgtl5000_arm2_vddio_reg_config,
 	},
 };
 
-struct platform_device sgtl5000_vddd_reg_devices = {
+static struct platform_device sgtl5000_arm2_vddd_reg_devices = {
 	.name	= "reg-fixed-voltage",
 	.id	= 2,
 	.dev	= {
-		.platform_data = &sgtl5000_vddd_reg_config,
+		.platform_data = &sgtl5000_arm2_vddd_reg_config,
 	},
 };
 
 #endif /* CONFIG_SND_SOC_SGTL5000 */
 
-
 static int imx6q_init_audio(void)
 {
 	struct clk *pll3_pfd, *esai_clk;
@@ -977,7 +976,11 @@ static int imx6q_init_audio(void)
 
 	clk_set_parent(esai_clk, pll3_pfd);
 	clk_set_rate(esai_clk, 101647058);
-
+#ifdef CONFIG_SND_SOC_SGTL5000
+	platform_device_register(&sgtl5000_arm2_vdda_reg_devices);
+	platform_device_register(&sgtl5000_arm2_vddio_reg_devices);
+	platform_device_register(&sgtl5000_arm2_vddd_reg_devices);
+#endif
 	return 0;
 }
 
diff --git a/arch/arm/mach-mx6/mx6_anatop_regulator.c b/arch/arm/mach-mx6/mx6_anatop_regulator.c
index 1bf2107..89c053a 100644
--- a/arch/arm/mach-mx6/mx6_anatop_regulator.c
+++ b/arch/arm/mach-mx6/mx6_anatop_regulator.c
@@ -320,12 +320,6 @@ static int __init regulators_init(void)
 	anatop_register_regulator(&vdd1p1_reg, ANATOP_VDD1P1, &vdd1p1_init);
 	anatop_register_regulator(&vdd3p0_reg, ANATOP_VDD3P0, &vdd3p0_init);
 
-#ifdef CONFIG_SND_SOC_SGTL5000
-	platform_device_register(&sgtl5000_vdda_reg_devices);
-	platform_device_register(&sgtl5000_vddio_reg_devices);
-	platform_device_register(&sgtl5000_vddd_reg_devices);
-#endif
-
 	return 0;
 }
 postcore_initcall(regulators_init);
-- 
1.7.7.4

